src/components/responsive-element.tsx

```diff
@@ -4,28 +4,49 @@ import classnames from "classnames";
 type ResponsiveElementTagName = "h1" | "h2" | "h3" | "h4" | "p";
 
 type ResponsiveElementInfo = {
-  desktop: ResponsiveElementTagName;
-  touch: ResponsiveElementTagName;
   children: React.ReactNode;
+  mobile?: ResponsiveElementTagName;
```

Comment:
hmm do we need all of these classes? i'm wondering if we could restrict ourselves to a subset--it feels to me like the default bulma classes have the possibility for weird conflicting behavior (if you specify an element type with "tabletOnly" and then another one for "touch" will you get duplicate elements?), i'm wondering if we could design our class/breakpoint structure to be less permissive but also less confusing/less potential for error, if that makes sense?

Reply:
Yeah good idea, there are a lot of them! What do you think of keeping only:

- mobile
- touch
- tablet
- desktop

Reply:
yeah those make sense! when one of these is specified, it means that device ONLY, correct? (eg if there was a responsive element with `mobile` and `desktop` props set, nothing would show up for tablet size?) that seems fine to me!

Reply:
no sorry, the [Bulma breakpoint](https://bulma.io/documentation/overview/responsiveness/#breakpoints) names are confusing. `mobile` is just mobile, `touch` is mobile/tablet, `tablet` is tablet size and up, and `desktop` is desktop size and up. 
I think that's usually what we might want though. I'm expecting we'll mostly use either `touch`/`desktop` or `mobile`/`tablet`.

<img width="1203" alt="image" src="https://user-images.githubusercontent.com/16906516/183770788-8f76f99e-9b5d-40c1-b28d-3ce772585a1d.png">

Reply:
hmmm ok so we COULD do something like.... 

````
ResponsiveElementInfo {
    ...
    tagNames: { mobile: ResponsiveElementTagName, tablet: ResponsiveElementTagName } | { touch: ResponsiveElementTagName, desktop: ResponsiveElementTagName }
}
````

which would kind of lock us into specific those specific splits but has stronger typing.... i'm kind of undecided haha, what do you think? not sure if this is too inflexible

Reply:
@shakao What do you think of using this stronger typing format, but adding an extra option to keep the possibility open for distinct tablet layouts, just in case. I can't imagine us needing any often configurations. So it'd be:

```
ResponsiveElementInfo {
    ...
    tagNames: { mobile: ResponsiveElementTagName, tablet: ResponsiveElementTagName }
      | { touch: ResponsiveElementTagName, desktop: ResponsiveElementTagName }
      | { mobile: ResponsiveElementTagName, tabletOnly: ResponsiveElementTagName, desktop: ResponsiveElementTagName }
}
```

Reply:
yeah i like this! small bonus of being more restrictive there's also fewer configurations to remember to test when we update things haha

Reply:
great, thanks so much for taking the time to work through this! I've made all those changes, so should be good for your final review.